### PR TITLE
Update Gatsby migration guide to use image imports.

### PR DESCRIPTION
**Pull request by leerob**
This aligns more with users migrating from Gatsby, who are used to importing local images in their repository. With image imports, you don't need to set the width and height - they are provided automatically.

https://nextjs.org/docs/basic-features/image-optimization#image-imports

### CODE REVIEW

File: docs/migrating/from-gatsby.md [hunk 1]
```diff
@@ -189,16 +189,20 @@ Instead, use the built-in [`next/image`](/docs/api-reference/next/image.md) comp
 
 ```jsx
 import Image from 'next/image'
+import profilePic from '../public/me.png'
 
-export default function Home() {
+function Home() {
```
Comment (leerob) [inline docs/migrating/from-gatsby.md:194]:
```suggestion
export default function Home() {
```

File: docs/migrating/from-gatsby.md [hunk 2]
```diff
@@ -189,16 +189,20 @@ Instead, use the built-in [`next/image`](/docs/api-reference/next/image.md) comp
...
-        src="/me.png"
+        src={profilePic}
         alt="Picture of the author"
-        width={500}
-        height={500}
+        // width={500} automatically provided
+        // height={500} automatically provided
+        // blurDataURL="data:..." automatically provided
+        // Optionally allows to add a blurred version of the image while loading
+        // placeholder="blur"
```
Comment (styfle) [inline docs/migrating/from-gatsby.md:205]: Lets show the gatsby equivalent props here so users can more easily migrate

For example `layout` would be important. I think gatsby `fixed` is the same as ours but `fluid` is probably `responsive` for next.

Also we should comment with the "blur up" placeholder equivalent for gatsby

Comment (leerob) [inline docs/migrating/from-gatsby.md:205]:
```suggestion
        // This is the same behavior as gatsby-image
        // placeholder="blur"
        // Gatsby's fixed prop equivalent
        // layout="fixed"
        // Gatsby's fluid prop equivalent
        // layout="responsive"
```

Comment (styfle) [reply docs/migrating/from-gatsby.md:205]:
```suggestion
        // When "responsive", similar to "fluid" from Gatsby
        // When "intrinsic", similar to "fluid" with maxWidth from Gatsby
        // When "fixed", similar to "fixed from "Gatsby"
        layout="responsive"
        // Optional, similar to "blur-up" from Gatsby
        placeholder="blur"
        // Optional, similar to width in graphql
        width={500}
        // Optional, similar to height in graphql
        height={500}
```

**Comment by kara** [comment]
This is awesome!  Do we also want to link to this new section from the [Image Optimization ](https://nextjs.org/docs/basic-features/image-optimization) docs? So users can find it if they start from image docs instead of Gatsby docs (maybe a 'migrating images from other x' section)

**Review by styfle — APPROVED** [review]

**Comment by styfle** [comment]
@kara Sounds good! Let’s add a link from @atcastle ’s PR

**Review by ijjk — APPROVED** [review]

**Pull request merged**